### PR TITLE
chore(jsutils): improved didYouMean to handle more edge cases

### PR DESCRIPTION
**Pull request by pranshuchittora**
### Description
Better edge cases handling for the `jsutils/didYouMean`. 

- Handling duplication
- Handling broken values like (`undefined`, `null`, `NaN`)

### Current behaviour
As of now, the module is not able to handle any duplicated values in the `Array`.
Similarly, if it encounters `undefined`, `null` etc then it converts it to the string "undefined", "null".

### Approach
I tried creating a new `Set` from the given `Array`, then finally filtering it in order to eliminate the broken or garbage values.

- [x] Tests Added

### CODE REVIEW

File: src/jsutils/__tests__/didYouMean-test.js [hunk 1]
```diff
@@ -35,4 +35,27 @@ describe('didYouMean', () => {
...
+      ' Did you mean "A", "B", or "C"?',
+    );
+  });
+  // Test for filtering null, undefined, NaN...
+  it('should filter the broken values', () => {
+    expect(didYouMean(['A', 'B', 'C', 'C', undefined, null])).to.equal(
+      ' Did you mean "A", "B", or "C"?',
+    );
+  });
+  it('Does accept an empty list', () => {
```
Comment (nveenjain) [inline src/jsutils/__tests__/didYouMean-test.js]: @pranshuchittora I don't think ```didYouMean``` function is defined for Numbers, hence NaN can't be passed... It only accepts string array, and all strings are required, and hence null/undefined can't be passed...

File: src/jsutils/didYouMean.js [hunk 2]
```diff
@@ -19,12 +19,16 @@ export default function didYouMean(firstArg, secondArg) {
       ? [firstArg, secondArg]
       : [undefined, firstArg];
 
+  const suggestionsArgFromSet = Array.from(new Set(suggestionsArg)).filter(
+    x => x,
+  );
+  console.log({ suggestionsArgFromSet });
```
Comment (IvanGoncharov) [inline src/jsutils/didYouMean.js]: It's an internal function that has Flow typings to accept only strings so this check is done by Flow linter.
Also, please review your PRs before submission and try to remove stuff like `console.log`.

File: src/jsutils/__tests__/didYouMean-test.js [hunk 3]
```diff
@@ -35,4 +35,27 @@ describe('didYouMean', () => {
       ' Did you mean the letter "A"?',
     );
   });
+  // Test for handling Duplications
+  it('Handle duplicates, resulting in single unique element', () => {
+    expect(didYouMean(['A', 'A'])).to.equal(' Did you mean "A"?');
+  });
+  it('Handle duplicates, resulting in two elements', () => {
```
Comment (IvanGoncharov) [inline src/jsutils/__tests__/didYouMean-test.js]: For future PRs: Please spend time and try to understand the code style of the file that you changing.
For example, other tests doesn't have comments before them but have an empty line between tests.

**Comment by IvanGoncharov** [comment]
@pranshuchittora Also please run `npm test` before submitting PRs.

File: src/jsutils/__tests__/didYouMean-test.js [hunk 4]
```diff
@@ -35,4 +35,27 @@ describe('didYouMean', () => {
...
+    expect(didYouMean(['A', 'B', 'A', 'B'])).to.equal(
+      ' Did you mean "A" or "B"?',
+    );
+  });
+  it('Handle duplicates, resulting in more thentwo elements', () => {
+    expect(didYouMean(['A', 'B', 'C', 'C', 'B', 'A'])).to.equal(
+      ' Did you mean "A", "B", or "C"?',
+    );
+  });
```
Comment (IvanGoncharov) [inline src/jsutils/__tests__/didYouMean-test.js]: This single test actually make sense so please extract it into separate PR

Comment (Cito) [reply src/jsutils/__tests__/didYouMean-test.js]: there is a typo: "more thentwo" -> "more than two"

Comment (IvanGoncharov) [reply src/jsutils/__tests__/didYouMean-test.js]: On second thought: It's the responsibility of `suggestionList` to remove duplicates so `didYouMean` should show duplicates as is.

File: src/jsutils/__tests__/didYouMean-test.js (hunk 1 above)
Comment (IvanGoncharov) [reply src/jsutils/__tests__/didYouMean-test.js]: @nveenjain Thanks for review 👍

File: src/jsutils/didYouMean.js (hunk 2 above)
Comment (pranshuchittora) [reply src/jsutils/didYouMean.js]: Pardon
